In [1]:
import pandas as pd

# Bring in CSV converted from PDF

# PDF: The World’s Religions in Figures: An Introduction to International Religious Demography, 
First Edition. Todd M. Johnson and Brian J. Grim 
© 2013 John Wiley & Sons, Ltd. Published 2013 by John Wiley & Sons, Ltd.

# Data source: Todd M. Johnson and Brian J. Grim, eds., World Religion Database (Leiden/Boston: Brill, accessed January 2012)

In [5]:
population1 = pd.read_csv('../data/johnson_grim/1910_2010_csv/population.csv')

In [6]:
population1.head(1)

Region Population 1910 Population 1970  Population 2010
0  Africa    124,541,000     365,450,000    1,022,234,000

In [7]:
agnostic1 = pd.read_csv('../data/johnson_grim/1910_2010_csv/agnostics.csv')

In [8]:
agnostic1.head(1)

Region A 1910  1910%   A 1970  1970%     A 2010  2010%
0  Africa  8,400    0.0  588,000    0.2  6,497,000    0.6

In [9]:
atheist1 = pd.read_csv('../data/johnson_grim/1910_2010_csv/atheist.csv')


In [10]:
atheist1.head(1)

Region Atheists 1910  1910% Atheists 1970  1970% Atheists 2010  2010%
0  Africa        1,100     0.0      103,000     0.0      571,000     0.1

In [11]:
buddhist1 = pd.read_csv('../data/johnson_grim/1910_2010_csv/buddhist.csv')


In [12]:
buddhist1.head(1)

Region Buddhists 1910  1910% Buddhists 1970  1970% Buddhists 2010  2010%
0  Africa         3,600     0.0        11,700     0.0       254,000     0.0

In [13]:
christian1 = pd.read_csv('../data/johnson_grim/1910_2010_csv/christian.csv')


In [14]:
christian1.head(1)

Region Christians 1910  1910% Christians 1970  1970% Christians 2010  2010%
0  Africa     11,636,000     9.3    137,867,000    37.7    494,053,000    48.3

In [15]:
hindu1 = pd.read_csv('../data/johnson_grim/1910_2010_csv/hindu.csv')

In [16]:
hindu1.head(1)

Region Hindus 1910  1910%  Hindus 1970  1970%  Hindus 2010  2010%
0  Africa    303,000     0.2   1,000,000     0.3   2,930,000     0.3

In [17]:
muslim1 = pd.read_csv('../data/johnson_grim/1910_2010_csv/muslim.csv')

In [18]:
muslim1.head(1)

Region  Muslims 1910  1910%   Muslims 1970  1970%   Muslims 2010  2010%
0  Africa   39,916,000    32.1   148,210,000    40.6   425,860,000    41.7

# Bring in CSV converted from PDF

# PDF: Projecting Global Religious Populations, 2020–50 from Journal of Religion and Demography 8 (2021) 124–153

# Data source: Todd M. Johnson and Brian J. Grim, eds., World Religion Database (Leiden/Boston: Brill, accessed July 2021).

In [22]:
population2 = pd.read_csv('../data/johnson_grim/2020_2050_csv/population.csv')

In [23]:
population2.head(1)

Region  Population 2020  Population 2050
0  Africa   1,340,598,000    2,489,275,000

In [24]:
agnostic2 = pd.read_csv('../data/johnson_grim/2020_2050_csv/agnostic.csv')

In [25]:
agnostic2.head(1)

Region Agnostics 2020  2020% Agnostics 2050  2050%
0  Africa     8,215,000     0.6    16,509,000     0.7

In [26]:
atheist2 = pd.read_csv('../data/johnson_grim/2020_2050_csv/atheist.csv')

In [27]:
atheist2.head(1)

Region     A 2020  2020%       A 2050  2050%
0  Africa   739,000     0.1   1,244,000     0.0

In [28]:
buddhist2 = pd.read_csv('../data/johnson_grim/2020_2050_csv/buddhist.csv')

In [29]:
buddhist2.head(1)

Region Buddhists 2020  2020% Buddhists 2050  2050%
0  Africa       365,000     0.0       611,000     0.0

In [30]:
christian2 = pd.read_csv('../data/johnson_grim/2020_2050_csv/christian.csv')

In [31]:
christian2.head(1)

Region Christians 2020  2020%  Christians 2050  2050%
0  Africa    654,913,000    48.9   1,280,641,000    51.4

In [32]:
hindu2 = pd.read_csv('../data/johnson_grim/2020_2050_csv/hindu.csv')

In [33]:
hindu2.head(1)

Region  Hindus 2020  2020%  Hindus 2050  2050%
0  Africa   3,555,000     0.3   5,031,000     0.2

In [34]:
muslim2 = pd.read_csv('../data/johnson_grim/2020_2050_csv/muslim.csv')

In [35]:
muslim2.head(1)

Region   Muslims 2020  2020%     Muslims 2050  2050%
0  Africa   556,923,000    41.5   1,043,699,000    41.9

# Merge 1910-2010 and 2020-2050 tables for each religion

In [37]:
population3 = pd.merge(population1, population2, left_on = 'Region', right_on = 'Region')

In [38]:
agnostic3 = pd.merge(agnostic1, agnostic2, left_on = 'Region', right_on = 'Region')

In [39]:
atheist3 = pd.merge(atheist1, atheist2, left_on = 'Region', right_on = 'Region')

In [40]:
buddhist3 = pd.merge(buddhist1, buddhist2, left_on = 'Region', right_on = 'Region')

In [41]:
christian3 = pd.merge(christian1, christian2, left_on = 'Region', right_on = 'Region')

In [42]:
hindu3 = pd.merge(hindu1, hindu2, left_on = 'Region', right_on = 'Region')

In [43]:
muslim3 = pd.merge(muslim1, muslim2, left_on = 'Region', right_on = 'Region')

# Bring in United Nations Demographic Indicators CSV

In [45]:
un_indicators = pd.read_csv('../data/indicators/un_indicators.csv')

C:\Users\graft\AppData\Local\Temp\ipykernel_99832\2938383926.py:1: DtypeWarning: Columns (2,3,4,7) have mixed types. Specify dtype option on import or set low_memory=False.
  un_indicators = pd.read_csv('../data/indicators/un_indicators.csv')


In [46]:
un_indicators.columns

Index(['SortOrder', 'LocID', 'Notes', 'ISO3_code', 'ISO2_code', 'SDMX_code',
       'LocTypeID', 'LocTypeName', 'ParentID', 'Location', 'VarID', 'Variant',
       'Time', 'TPopulation1Jan', 'TPopulation1July', 'TPopulationMale1July',
       'TPopulationFemale1July', 'PopDensity', 'PopSexRatio', 'MedianAgePop',
       'NatChange', 'NatChangeRT', 'PopChange', 'PopGrowthRate',
       'DoublingTime', 'Births', 'Births1519', 'CBR', 'TFR', 'NRR', 'MAC',
       'SRB', 'Deaths', 'DeathsMale', 'DeathsFemale', 'CDR', 'LEx', 'LExMale',
       'LExFemale', 'LE15', 'LE15Male', 'LE15Female', 'LE65', 'LE65Male',
       'LE65Female', 'LE80', 'LE80Male', 'LE80Female', 'InfantDeaths', 'IMR',
       'LBsurvivingAge1', 'Under5Deaths', 'Q5', 'Q0040', 'Q0040Male',
       'Q0040Female', 'Q0060', 'Q0060Male', 'Q0060Female', 'Q1550',
       'Q1550Male', 'Q1550Female', 'Q1560', 'Q1560Male', 'Q1560Female',
       'NetMigrations', 'CNMR'],
      dtype='object')

In [47]:
un_indicators = un_indicators[['Time', 'Location', 'LocTypeName', 'ISO3_code', 'Births', 'CBR', 'TFR', 'NRR', 'PopGrowthRate', 'TPopulation1Jan']]

In [48]:
un_indicators

Time                           Location   LocTypeName ISO3_code  \
0      1950  ADB region: Central and West Asia           NaN       NaN   
1      1951  ADB region: Central and West Asia           NaN       NaN   
2      1952  ADB region: Central and West Asia           NaN       NaN   
3      1953  ADB region: Central and West Asia           NaN       NaN   
4      1954  ADB region: Central and West Asia           NaN       NaN   
...     ...                                ...           ...       ...   
84355  2097          Wallis and Futuna Islands  Country/Area       WLF   
84356  2098          Wallis and Futuna Islands  Country/Area       WLF   
84357  2099          Wallis and Futuna Islands  Country/Area       WLF   
84358  2100          Wallis and Futuna Islands  Country/Area       WLF   
84359  2101          Wallis and Futuna Islands  Country/Area       WLF   

         Births     CBR     TFR     NRR  PopGrowthRate  TPopulation1Jan  
0      2933.594  42.385  5.9875  1.6659          2.058        68501.280  
1      3012.920  42.626  5.9815  1.7067          2.143        69925.568  
2      3099.971  42.913  5.9847  1.7433          2.212        71439.952  
3      3193.546  43.228  5.9954  1.7815          2.271        73037.479  
4      3288.272  43.495  6.0042  1.8136          2.342        74715.250  
...         ...     ...     ...     ...            ...              ...  
84355     0.046   6.458  1.5266  0.7377         -1.081            7.161  
84356     0.046   6.542  1.5249  0.7370         -1.164            7.084  
84357     0.046   6.658  1.5266  0.7379         -1.250            7.002  
84358     0.047   6.771  1.5271  0.7382         -1.091            6.915  
84359       NaN     NaN     NaN     NaN            NaN            6.840  

[84360 rows x 10 columns]

# Break UN Indicators into multiple tables: World, Region

In [50]:
world = un_indicators.loc[un_indicators['Location'] == 'World']

In [51]:
world

Time Location LocTypeName ISO3_code      Births     CBR     TFR  \
35568  1950    World       World       NaN   91823.936  36.831  4.8519   
35569  1951    World       World       NaN   92507.128  36.464  4.8164   
35570  1952    World       World       NaN   97371.459  37.681  5.0008   
35571  1953    World       World       NaN   97291.452  36.935  4.9223   
35572  1954    World       World       NaN  100187.854  37.301  4.9978   
...     ...      ...         ...       ...         ...     ...     ...   
35715  2097    World       World       NaN  111397.100  10.905  1.8493   
35716  2098    World       World       NaN  110941.160  10.872  1.8454   
35717  2099    World       World       NaN  110500.810  10.841  1.8419   
35718  2100    World       World       NaN  110028.472  10.808  1.8380   
35719  2101    World       World       NaN         NaN     NaN     NaN   

          NRR  PopGrowthRate  TPopulation1Jan  
35568  1.6316          1.738     2.471424e+06  
35569  1.6377          1.747     2.514762e+06  
35570  1.7256          1.935     2.559092e+06  
35571  1.7133          1.900     2.609080e+06  
35572  1.7612          1.993     2.659132e+06  
...       ...            ...              ...  
35715  0.8843         -0.105     1.022089e+07  
35716  0.8826         -0.112     1.021021e+07  
35717  0.8812         -0.119     1.019877e+07  
35718  0.8795         -0.127     1.018661e+07  
35719     NaN            NaN     1.017371e+07  

[152 rows x 10 columns]

# Checking and correcting discrepancies between Johnson_Grim religion data and UN data continent categorization

Johnson_Grim uses 'Latin America', but UN data splits into 'Americas' and 'Latin America and the Caribbean'

In [53]:
geographic_region = un_indicators.loc[un_indicators['LocTypeName'] == 'Geographic region']

In [54]:
geographic_region['Location'].value_counts()

Location
Africa                             152
Asia                               152
Europe                             152
Americas                           152
Latin America and the Caribbean    152
Northern America                   152
Oceania                            152
Name: count, dtype: int64

In [55]:
americas = geographic_region.loc[geographic_region['Location'] == 'Americas']

In [56]:
Latin_America_and_the_Caribbean = geographic_region.loc[geographic_region['Location'] == 'Latin America and the Caribbean']

In [57]:
americas_latin_merge = pd.merge(americas, Latin_America_and_the_Caribbean, left_on = 'Time', right_on = 'Time')

In [58]:
americas_latin_merge['Births'] = americas_latin_merge['Births_x'] + americas_latin_merge['Births_y']

In [59]:
americas_latin_merge['CBR'] = americas_latin_merge['CBR_x'] + americas_latin_merge['CBR_y']

In [60]:
americas_latin_merge['TFR'] = americas_latin_merge['TFR_x'] + americas_latin_merge['TFR_y']

In [61]:
americas_latin_merge['NRR'] = americas_latin_merge['NRR_x'] + americas_latin_merge['NRR_y']

In [62]:
americas_latin_merge['TPopulation1Jan'] = americas_latin_merge['TPopulation1Jan_x'] + americas_latin_merge['TPopulation1Jan_y']

In [63]:
americas_latin_merge['PopGrowthRate'] = round(((americas_latin_merge['TPopulation1Jan'].pct_change()) * 100), 3)

In [64]:
americas_latin_merge['Location'] = 'Latin America'

In [65]:
latin_america = americas_latin_merge[['Time', 'Location', 'Births', 'CBR', 'TFR', 'NRR', 'PopGrowthRate', 'TPopulation1Jan']]

In [66]:
latin_america

Time       Location     Births     CBR      TFR     NRR  PopGrowthRate  \
0    1950  Latin America  18669.028  77.381  10.1568  3.7923            NaN   
1    1951  Latin America  19165.035  77.563  10.2264  3.8506          2.247   
2    1952  Latin America  19646.510  77.626  10.2933  3.9086          2.286   
3    1953  Latin America  20093.248  77.539  10.3499  3.9628          2.312   
4    1954  Latin America  20606.443  77.581  10.4325  4.0288          2.332   
..    ...            ...        ...     ...      ...     ...            ...   
147  2097  Latin America  14842.506  17.145   3.2903  1.5978         -0.383   
148  2098  Latin America  14767.885  17.123   3.2887  1.5972         -0.383   
149  2099  Latin America  14703.652  17.111   3.2893  1.5977         -0.388   
150  2100  Latin America  14628.260  17.090   3.2877  1.5970         -0.388   
151  2101  Latin America        NaN     NaN      NaN     NaN         -0.394   

     TPopulation1Jan  
0         497980.009  
1         509167.297  
2         520808.579  
3         532850.652  
4         545278.842  
..               ...  
147      1725134.718  
148      1718534.988  
149      1711868.764  
150      1705218.645  
151      1698502.295  

[152 rows x 8 columns]

In [67]:
geographic_region2 = pd.concat([geographic_region, latin_america], axis=0, ignore_index=True).reset_index()

In [68]:
geographic_region2 = geographic_region[((geographic_region['Location'] != 'Americas') & (geographic_region['Location'] != 'Latin America and the Caribbean'))]

In [69]:
geographic_region2['Location'].value_counts()

Location
Africa              152
Asia                152
Europe              152
Northern America    152
Oceania             152
Name: count, dtype: int64

In [70]:
latin_america

Time       Location     Births     CBR      TFR     NRR  PopGrowthRate  \
0    1950  Latin America  18669.028  77.381  10.1568  3.7923            NaN   
1    1951  Latin America  19165.035  77.563  10.2264  3.8506          2.247   
2    1952  Latin America  19646.510  77.626  10.2933  3.9086          2.286   
3    1953  Latin America  20093.248  77.539  10.3499  3.9628          2.312   
4    1954  Latin America  20606.443  77.581  10.4325  4.0288          2.332   
..    ...            ...        ...     ...      ...     ...            ...   
147  2097  Latin America  14842.506  17.145   3.2903  1.5978         -0.383   
148  2098  Latin America  14767.885  17.123   3.2887  1.5972         -0.383   
149  2099  Latin America  14703.652  17.111   3.2893  1.5977         -0.388   
150  2100  Latin America  14628.260  17.090   3.2877  1.5970         -0.388   
151  2101  Latin America        NaN     NaN      NaN     NaN         -0.394   

     TPopulation1Jan  
0         497980.009  
1         509167.297  
2         520808.579  
3         532850.652  
4         545278.842  
..               ...  
147      1725134.718  
148      1718534.988  
149      1711868.764  
150      1705218.645  
151      1698502.295  

[152 rows x 8 columns]